File: app/src/main/java/com/jameskbride/eventbusmvpdemo/network/NetworkErrorViewPresenter.java
package com.jameskbride.eventbusmvpdemo.network;

import com.jameskbride.eventbusmvpdemo.bus.BusAware;
import com.jameskbride.eventbusmvpdemo.bus.NetworkRequestEvent;

import org.greenrobot.eventbus.EventBus;

public class NetworkErrorViewPresenter extends BusAware {

    NetworkErrorView view;

    public NetworkErrorViewPresenter(EventBus bus) {
        super(bus);
    }

    public void retry(NetworkRequestEvent networkRequestEvent) {
        bus.post(networkRequestEvent);
        view.dismiss();
    }

    public interface NetworkErrorView {
        void dismiss();
    }
}

File: app/src/main/java/com/jameskbride/eventbusmvpdemo/network/NetworkErrorViewPresenter.kt
package com.jameskbride.eventbusmvpdemo.network

import com.jameskbride.eventbusmvpdemo.bus.NetworkRequestEvent
import org.greenrobot.eventbus.EventBus
import com.jameskbride.eventbusmvpdemo.bus.BusAware


class NetworkErrorViewPresenter constructor(override val eventBus: EventBus) : BusAware {

    lateinit var view: NetworkErrorView

    fun retry(networkRequestEvent: NetworkRequestEvent) {
        eventBus.post(networkRequestEvent)
        view.dismiss()
    }
}

interface NetworkErrorView {
    fun dismiss()
}